src/payment-service.ts

```diff
@@ -226,20 +218,52 @@ export async function handlePaymentQuerystring(
     )
 }
 
-export async function handlePaymentSubmissionEvent({
-  formSubmissionResult,
-  paymentSubmissionEvent,
-  paymentReceiptUrl,
-}: {
-  formSubmissionResult: SubmissionTypes.FormSubmissionResult
-  paymentSubmissionEvent: SubmissionEventTypes.PaymentSubmissionEvent
-  paymentReceiptUrl: string
-}): Promise<SubmissionTypes.FormSubmissionResult | undefined> {
-  console.log('Attempting to handle submission with payment submission event')
-  const { definition: form, submission } = formSubmissionResult
+export function checkForPaymentSubmissionEvent(formSubmission: FormSubmission):
+  | {
+      paymentSubmissionEvent: SubmissionEventTypes.PaymentSubmissionEvent
+      amount: number
+    }
+  | undefined {
+  const submissionEvents = formSubmission.definition.submissionEvents || []
+  const paymentSubmissionEvent = submissionEvents.reduce(
+    (
+      p: SubmissionEventTypes.PaymentSubmissionEvent | null,
+      submissionEvent,
+    ) => {
+      if (p) {
+        return p
+      }
+      if (
+        (submissionEvent.type === 'CP_PAY' ||
+          submissionEvent.type === 'BPOINT' ||
+          submissionEvent.type === 'WESTPAC_QUICK_WEB') &&
```

Comment:
Can we use the `toPaymentSubmissionEvent` function in the `sdk-core` `typeCastService` to do this? Trying to consolidate logic like this all in one place so when we add new things we don't forget to changes it in 100 places

Reply:
All good otherwise

Reply:
Love it!!!